File: aula5/parte1/python/mago.py
from personagem import Personagem


class Mago(Personagem):
    def __init__(self, nome: str, vida: int, nivel: int):
        super().__init__(nome, vida, nivel)
        self.mana = 50

    @property
    def mana(self) -> int:
        return self._mana

    @mana.setter
    def mana(self, valor: int) -> None:
        if valor < 0:
            raise ValueError("A mana não pode ser negativa.")
        self._mana = valor

    def habilidade(self) -> str:
        return "rajada arcana"

    def ficha(self) -> None:
        super().ficha()
        print(f"Mana:  {self._mana}")

File: aula5/parte1/java/Mago.java
public class Mago extends Personagem {

    private int mana;

    public Mago(String nome, int vida, int nivel) {
        super(nome, vida, nivel);
        setMana(50);
    }

    public int getMana() {
        return this.mana;
    }

    public void setMana(int mana) {
        if (mana < 0) {
            throw new IllegalArgumentException("A mana não pode ser negativa.");
        }
        this.mana = mana;
    }

    @Override
    public String habilidade() {
        return "rajada arcana";
    }

    @Override
    public void ficha() {
        super.ficha();
        System.out.println("Mana:  " + this.mana);
    }
}
